esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -1320,6 +1351,138 @@ void EsdlServiceImpl::getSoapError( StringBuffer& out,
     out.clear().append(finger-start,start);
 }
 
+void EsdlServiceImpl::adjustTargetConfig(Owned<IPTree>& tgtcfg) const
+{
+    static const VStringBuffer gwLocalSecretXPath("Gateways/Gateway[@url='%s*']", gwLocalSecretPrefix);
+
+    if (!tgtcfg)
+        return;
+
+    if (tgtcfg->hasProp(gwLocalSecretXPath))
+    {
+        tgtcfg.setown(createPTreeFromIPT(tgtcfg));
+        const char* qt = tgtcfg->queryProp("@querytype");
+        bool published = (qt && (strieq(qt, "roxie") || strieq(qt, "wsecl")));
+        const char* permission = (published ? "allowPublishedGatewayUsage" : "allowUnpublishedGatewayUsage");
+        Owned<IPTreeIterator> gwsToResolve(tgtcfg->getElements(gwLocalSecretXPath));
+        ForEach(*gwsToResolve)
+            resolveGatewayLocalSecret(gwsToResolve->query(), permission);
+    }
+}
+
+void EsdlServiceImpl::resolveGatewayLocalSecret(IPTree& gateway, const char* permission) const
+{
+    const char* name = gateway.queryProp("@name");
+    StringBuffer url(gateway.queryProp("@url"));
+    if (url.isEmpty())
+        throw makeStringExceptionV(-1, "gateway %s: missing url property", name);
+    if (strncmp(url, gwLocalSecretPrefix, gwLocalSecretPrefixLength))
+        throw makeStringExceptionV(-1, "gateway %s: expected '%s...'; got '%s'", name, gwLocalSecretPrefix, url.str());
+    const char* identification = url.str() + gwLocalSecretPrefixLength;
```

Comment:
For ECL I currently require connection secrets to start with "http-connect-".  We should consider doing the same here.  It means the secrets were explictly created for this purpose.  I may regret it in the future, who knows, but it gives a little control and forced standardization.